### DISCUSSION TITLE
My "nuxt instance unavailable" message

### DISCUSSION THREAD

**Discussion by daniluk4000** [nuxt/nuxt#16292] [General]
Hey guys. 

I've been trying to use Nuxt 3 for a while on side project. Now, I've tried to migrate my own, very big project to Nuxt Bridge.

I've been dealing with multiple issues during my migration: plugins not working, server middlewares needs updating, e.t.c.
But one thing was there all this time. "nuxt instance unavailable". 

Typical use case from Nuxt 2:
- You have your asyncData
- You have Vuex actions
- You have some helpers from injected methods
- asyncData, injected methods and actions have async/await in it

Also, you can have:
- Plugin
- Server middleware
- etc

I was able to use as many awaits as I wanted. Nuxt 3 inproduced new change: your context is lost after single await in setup function. Nuxt team said it's because of how Vue works and I have to use script setup sugar to deal with it. I used it and context was still lost. I had to put some functions to CSR from asyncData as I NEED to make multiple await calls and I CAN'T use Promise.all for it.

Today, I got to know you'll get the same error everywhere where you have access in Nuxt App - for example, in Plugin. 
Does plugin have setup function? I think not.

I've read many issues about it. Both community and maintainers (@pi0 @danielroe) claim this as a feature. This discussion is to ask, will this feature remain here forever or do you guys have any plans to change this behavior in some future? 

I'm afraid we'll have to move from Nuxt to something else if this single error will not be resolved (yet I don't see how we'll even make it). It makes it impossible to write plugins, middlewares, and even asyncData outside setup. Nuxt is the best and we love it, we wrote almost 500 components on it and it's groving. But I don't know how will we ever be able to migrate from Nuxt 2.

Possible workaround I see is to create middleware, plugin and component for each async function or move every second and more await callc to CSR (which makes some of our ssr features meaningless). Both solutions I consider as bad practice and basically a workaround for some bug, not a feature.

**Comment by vedmant** [comment]
I'm also having hard time with Nuxt 3, and thinking that I may stay with Nuxt 2 for foreseeable future.

**Comment by TheAlexLichter** [comment]
I assume you have read https://github.com/nuxt/framework/issues/5740#issuecomment-1229197529?

**Comment by vedmant** [comment]
@manniL Yes, I read that but the issue with migration from Nuxt 2 is that context was always available, which is not the case now. Also when migrating I'm keeping options API, and looks like Nuxt 3 basically phased out options API, there is no documentation or examples how to use Nuxt 3 with options API, I generally have no idea should I use compostables in options API or and where or how in component lifecycle I can use them.

**Reply by TheAlexLichter** [reply]
I agree, options API is not well covered. Feel free to raise an issue about that.

You should not use composables in the Options API but the "bare options". useFetch -> `$fetch` for example.
